### ISSUE TITLE
Boa's start-up time getting hit with GC Allocation

### ISSUE THREAD

**Issue by jasonwilliams** [boa-dev/boa#3896]
Boa spends a large proportion of time in realm creation comparitive to actually running the code. 
It seems a lot of this is due to building out the global objects on the heap and storing them causing the GC to do multiple collections before the program has even started.

We may wish to look at some sort of snapshot for built-ins or staticlly storing them somewhere (switch of GC for built-ins)

You can load the below profiler into [ui.perfetto.dev](https://ui.perfetto.dev/) or chrome developer tools and take a look. 
![image](https://github.com/boa-dev/boa/assets/936006/a7a178d6-3d6a-47b9-97a0-38d150c94779)

[chrome_profiler.json](https://github.com/user-attachments/files/16120205/chrome_profiler.json)

**Comment by nekevss** [comment]
Raising the threshold at the below might help.

https://github.com/boa-dev/boa/blob/fc2a6e09969772feba98eaa89aaf89ca4797e925/core/gc/src/lib.rs#L66

**Comment by raskad** [comment]
> Raising the threshold at the below might help.
> 
> https://github.com/boa-dev/boa/blob/fc2a6e09969772feba98eaa89aaf89ca4797e925/core/gc/src/lib.rs#L66

Yeah I also think we should raise the default threshold and make it configurable from outside of `boa_gc`.

**Comment by jasonwilliams** [comment]
```
Hello world!
undefined
dhat: Total:     611,648 bytes in 3,019 blocks
dhat: At t-gmax: 382,386 bytes in 2,283 blocks
dhat: At t-end:  354,488 bytes in 2,258 blocks
```

Also I've noticed that in DHAT we spend a lot of time growing the vector for property_table.
![image](https://github.com/boa-dev/boa/assets/936006/8dfa73ac-cea0-478e-8b80-10a4fb8a4ce2)

If we know ahead of time how many static methods are on a builtin, we should be passing this number through so it can create the `vec::with_capacity(n)`

**Comment by jasonwilliams** [comment]
We’re not able to set each property_table with a capacity right now due to the fact that for each template the inner shape [is cloned](https://github.com/boa-dev/boa/blob/main/core/engine/src/object/shape/shared_shape/template.rs#L25) from the first shape. 

We may need to set new root shapes [here](https://github.com/boa-dev/boa/blob/main/core/engine/src/context/intrinsics.rs#L1397), each one having its own set of prototype storage capacity and property capacity.

@HalidOdat thoughts?
